package/root/usr/local/sbin/resize_rootfs.sh

```diff
@@ -1,13 +1,24 @@
 #!/bin/bash
 
-set -xe
+if [ $# -eq 0 ]; then
+	echo "No arguments provided... please specify device to resize!"
+	echo "e.g. $(basename $0) /dev/mmcblk1"
+	exit 1
+fi
+
+if [ ! -b "$1" ]; then
+	echo "Specified device '$1' does not exist! Abort!"
+	exit 1
+fi
 
 if [ "$(id -u)" -ne "0" ]; then
 	echo "This script requires root."
 	exit 1
 fi
 
```

Comment:
If we do that, don't we start getting into the realm of putting further effort into to to pretty it up, and also make it so that it is harder to use in an automated workflow? At the moment, I could add it to the config script I use to provision a new image if I wanted to, and it would have zero input requirements.